src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+function deleteRedundantNumber(numberWithRedundantNumber: number): number {
+  return Math.floor(numberWithRedundantNumber * 100) / 100;
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const maxSide = Math.max(this.a, this.b, this.c);
+
+    if (
+      [this.a, this.b, this.c].some((item) => item <= 0) ||
+      maxSide >= this.a + this.b + this.c - maxSide
+    ) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return deleteRedundantNumber(area);
+  }
+}
+
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('your error message');
+    }
+  }
 
-export class Circle implements Figure {}
+  getArea(): number {
+    const area = Math.PI * Math.pow(this.radius, 2);
+
+    return deleteRedundantNumber(area);
+  }
+}
 
-export class Rectangle implements Figure {}
+export class Rectangle implements Figure {
+  public shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('your error message');
```

Comment:
[CODE STYLE] - The error message in the constructor of the Rectangle class is not descriptive. It would be more helpful to specify which condition failed, for example, 'Rectangle width and height must be positive numbers.'